src/charm_state.py

```diff
@@ -202,3 +223,18 @@ def flask_port(self) -> int:
             The port number to use for the Flask server.
         """
         return 8000
+
+
+KNOWN_CHARM_CONFIG = (
```

Comment:
Do you think these should be read from `config.yaml`?

Reply:
Since the user-defined configurations are added to the `config.yaml` during the build time, reading from `config.yaml` during the runtime will include the user-defined configurations too.

Reply:
Do you have any ideas for how we could reduce the duplication of information? Or manage the risk of divergence?

Reply:
There's a unit test `test_known_charm_config` that will check the consistency between the `KNOW_CHARM_CONFIG` and `config.yaml`.